src/lib/mina_graphql/types.ml

```diff
@@ -356,11 +356,12 @@ module Itn = struct
               |> Unsigned.UInt16.of_int )
         ; field "peerId"
             ~args:Arg.[]
-            ~doc:"Peer id" ~typ:(non_null string)
+            ~doc:"Peer id" ~typ:string
             ~resolve:(fun { ctx = (_ : bool), mina; _ } _ ->
               Mina_lib.config mina
               |> fun Mina_lib.Config.{ gossip_net_params; _ } ->
-              Mina_net2.Keypair.to_peer_id gossip_net_params.keypair )
+              Option.map ~f:Mina_net2.Keypair.to_peer_id
```

Comment:
Hmm, I approved, but thinking on this some more... we should probably return the keypair that was generated inside of gossip_net library, so that way you can always determine the peer id of a node via the graphql, even if it's dynamically generated.

Reply:
This is the code as formerly on compatible. I'll try though.

Reply:
If it was what was there before, and it's too hard to implement it the way I suggested, feel free to drop it.

Reply:
I've added a hack that updates the config with the new value (since it was already mutable, presumably for this purpose). Note though that I misspoke in my comment above: this is an ITN GraphQL type which we will likely never use again :)